plugins/nf-ga4gh/src/main/nextflow/ga4gh/tes/executor/TesExecutor.groovy

```diff
@@ -158,6 +160,14 @@ class TesExecutor extends Executor implements ExtensionPoint {
         log.debug "[TES] Launching process > ${task.name} -- work folder: ${task.workDir}"
         new TesTaskHandler(task, this)
     }
+
+    protected BashWrapperBuilder createBashWrapperBuilder(TaskRun task) {
+        // creates the wrapper script
+        final builder = new BashWrapperBuilder(task)
+        // job directives headers
+        builder.headerScript = "NXF_CHDIR=${Escape.path(task.workDir)}"
+        return builder
+    }
```

Comment:
This should be done here instead: https://github.com/nextflow-io/nextflow/blob/048c3955236cb7ff1233323c55554c0b4e966bb0/plugins/nf-ga4gh/src/main/nextflow/ga4gh/tes/executor/TesTaskHandler.groovy#L170-L172

TesBashBuilder is a subclass of BashWrapperBuilder which is specialized for TES, so something like this should do:
```groovy
    protected TesBashBuilder newTesBashBuilder(TaskRun task, String remoteBinDir) {
        final builder = new TesBashBuilder(task, remoteBinDir)
        builder.headerScript = "NXF_CHDIR=${Escape.path(task.workDir)}"
        return builder
    }
```

Reply:
Good call will fix